encoder/vaapiencoder_vp9.cpp

```diff
@@ -159,9 +157,7 @@ YamiStatus VaapiEncoderVP9::doEncode(const SurfacePtr& surface,
 
     PicturePtr picture(new VaapiEncPictureVP9(m_context, surface, timeStamp));
 
-    m_frameCount %= keyFramePeriod();
-    picture->m_type = (m_frameCount ? VAAPI_PICTURE_P : VAAPI_PICTURE_I);
-    m_frameCount++;
+    picture->m_type = (forceKeyFrame ? VAAPI_PICTURE_I : VAAPI_PICTURE_P);
```

Comment:
Reset gop or not will depends on the implementation. However, here are some think about this:
Suppose key period is 30, User request key frame at 29. If we do not reset the GOP, it will generate a key frame in 29 and 30, this is unnecessary and will make bitrate very high, at this point.
Base on this we choose to reset GOP at h264 encoder